ledger/simulation/simulator.go

```diff
@@ -111,30 +112,27 @@ var proxySigner = crypto.PrivateKey{
 // check verifies that the transaction is well-formed and has valid or missing signatures.
 // An invalid transaction group error is returned if the transaction is not well-formed or there are invalid signatures.
 // To make things easier, we support submitting unsigned transactions and will respond whether signatures are missing.
-func (s Simulator) check(hdr bookkeeping.BlockHeader, txgroup []transactions.SignedTxn, debugger logic.EvalTracer) ([]int, error) {
+func (s Simulator) check(hdr bookkeeping.BlockHeader, txgroup []transactions.SignedTxn, signaturesOptional bool, tracer logic.EvalTracer) error {
```

Comment:
Good idea, but I don't want to change the tracer argument from `logic.EvalTracer` to `*evalTracer`. Because it's defined as the more generic tracer, the test `TestSimulateWithTrace` is able to work. This test uses the mock tracer to ensure our simulated environment still calls tracer hooks properly.

I did something similar to what you suggested in ea3372ea0 though, which uses the `ResultEvalOverrides` struct instead of `signaturesOptional bool` to pass this option. Let me know what you think.

Reply:
ouch, didn't see it coming, I thought `s.simulateWithTracer` is not taking interface, now it seems a bit involved...